pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -534,6 +558,52 @@ impl ImageLayerInner {
         Ok(planner.finish())
     }
 
+    /// Given a key range and filter function, generate a read plan and execute it, then for
+    /// each value we read write it onwards into the writer.
+    pub(super) async fn do_reads_and_write<F>(
```

Comment:
nit: Can we give this a different name since it's part of the public interface? Something like `rewrite_layer` or `filter_layer`.

Reply:
Hmm as far as unifying with my work goes, I kinda like the `filter_layer` or just `filter` (because we are already calling it on a layer) makes sense.